week3/homework/src/index.js

```diff
@@ -1,3 +1,73 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+const express = require('express');
+const uuid    = require('uuid/v4');
+
+const { readFile: _readFile, writeFile: _writeFile } = require('fs');
+const { promisify } = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf8').then(
+    JSON.parse,
+    () => ({})
+  );
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos, null, 2));
+}
+
+const app = express();
+app.use(express.json());
+
+// Add todo to the file
+app.post('/todos', async(request, response) => {
+  const newTodo = request.body;
+  newTodo.done = false;
+  const todos = await readTodos();
+  todos[uuid()] = newTodo;
+  await writeTodos(todos);
+  response.json(todos);
+});
+
+// Get todo by ID  'readTodo'
+app.get('/todos/:id', async(request, response) => {
+  const todos = await readTodos();
+  const id = request.params.id;
+  await writeTodos(todos);
+ id in todos ? response.json(todos[id]) : response.json('This todo does not exist!') ;
```

Comment:
You should return an object like so: `{ error:  'This todo does not exist!' }`. This will become important when you'll start looking into front-end stuff for instance.